packages/nx/src/command-line/reset/reset.ts

```diff
@@ -1,23 +1,166 @@
 import { rmSync } from 'fs-extra';
 import { daemonClient } from '../../daemon/client/client';
-import {
-  cacheDir,
-  projectGraphCacheDirectory,
-} from '../../utils/cache-directory';
+import { cacheDir, workspaceDataDirectory } from '../../utils/cache-directory';
 import { output } from '../../utils/output';
+import { prompt } from 'enquirer';
+import { ResetCommandOptions } from './command-object';
+import * as chalk from 'chalk';
 
-export async function resetHandler() {
+// Wait at max 5 seconds before giving up on a failing operation.
+const INCREMENTAL_BACKOFF_MAX_DURATION = 5000;
+
+// If an operation fails, wait 100ms before first retry.
+const INCREMENTAL_BACKOFF_FIRST_DELAY = 100;
+
+export async function resetHandler(args: ResetCommandOptions) {
+  if (
+    args.all === undefined &&
+    args.cache === undefined &&
+    args.daemon === undefined &&
+    args.workspaceData === undefined
+  ) {
+    const { resetOptions } = await prompt<{
+      resetOptions: (keyof ResetCommandOptions)[];
+    }>({
+      type: 'multiselect',
+      name: 'resetOptions',
+      message: 'What would you like to reset?',
+      choices: [
+        { name: 'all', message: 'Everything' },
+        { name: 'cache', message: 'Nx Cache' },
+        { name: 'daemon', message: 'Nx Daemon' },
+        { name: 'workspaceData', message: 'Workspace Data' },
+      ],
+      footer: 'Press <space> to select, <return> to submit.',
+      indicator(state, choice) {
```

Comment:
This is cool. Let's change the indicator in other places.

But when I reset, I don't want to be bothered by a prompt. I just want things reset.

We can keep the granularity flags but let's get rid of the prompt.